server/api.go

```diff
@@ -853,3 +847,163 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request, userID s
+		PostId string `json:"post_id"`
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		mlog.Error("Error decoding JSON body", mlog.Err(err))
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Title == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Repo == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repo name.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.PostId == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a postID", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	ghIssue := &github.IssueRequest{
+		Title: &issue.Title,
+		Body:  &issue.Body,
+	}
+
+	permalink := p.getPermaLink(issue.PostId)
+
+	mmMessage := fmt.Sprintf("_Issue created from a [message in Mattermost](%v)_.", permalink)
+	if len(*ghIssue.Body) > 0 {
+		mmMessage = fmt.Sprintf("\n\n_Issue created from a [message in Mattermost](%v)_.", permalink)
+	}
+	*ghIssue.Body = *ghIssue.Body + mmMessage
+
+	// Make sure user have a connected github account
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	currentUser, appErr := p.API.GetUser(userID)
+	if appErr != nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load current user", StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	splittedRepo := strings.Split(issue.Repo, "/")
+	owner := splittedRepo[0]
+	repoName := splittedRepo[1]
+
+	githubClient := p.githubConnect(*info.Token)
+	result, resp, apiErr := githubClient.Issues.Create(context.Background(), owner, repoName, ghIssue)
+	if apiErr != nil {
+		writeAPIError(w,
+			&APIErrorResponse{
+				ID: "",
+				Message: "failed to create issue: " + getFailReason(resp.StatusCode,
+					issue.Repo,
+					currentUser.Username),
+				StatusCode: resp.StatusCode,
+			})
+		return
+	}
+
+	if resp.Response.StatusCode == http.StatusGone {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Issues are disabled on this repository.", StatusCode: http.StatusMethodNotAllowed})
+		return
+	}
+
+	post, appErr := p.API.GetPost(issue.PostId)
+	if appErr != nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + issue.PostId, StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + issue.PostId + ": not found", StatusCode: http.StatusNotFound})
+		return
+	}
+
+	rootId := issue.PostId
+	if post.RootId != "" {
+		rootId = post.RootId
+	}
+
+	reply := &model.Post{
+		Message:   fmt.Sprintf("Created GitHub issue [#%v](%v)", *result.Number, *result.HTMLURL),
```

Comment:
@jfrerich Maybe we can make it more clear _what_ message what used to create the issue. Also is it needed to mention that it's a GitHub issue?
```suggestion
		Message:   fmt.Sprintf("Created [#%v](%v) from a [message](%s)", *result.Number, *result.HTMLURL, permalink),
```

Reply:
@hanzei, I think it might be useful to recognize this is a GitHub issue.  Although a user can decipher it from the link and will know that they created the issue in Github, I consider it subtle advertising for users scanning the posts in a channel

Reply:
Also, as more plugins are added, that create issues, it might be useful to acknowledge in the message where the issue went..  (Examples: github, jira, gitlab, bitbucket, .etc)

Reply:
Got, makes sense to keep mentioning that it's about a GitHub issue. What do you think about the message link?

Reply:
I think it is ok to exclude the message link also.  The post is a reply to the original post, creating a thread.  I'm 0/5 though
![image](https://user-images.githubusercontent.com/7575921/81304857-3d11b280-9043-11ea-8fe1-f00f175c4190.png)

Reply:
I did the link given that it doesn't have to be the root post that was used for the issue.